shared/types/submission.ts

```diff
@@ -80,6 +80,39 @@ export type StorageModeSubmissionBase = z.infer<
  * Multirespondent submission typings as stored in the database.
  */
 
+export enum WorkflowStatus {
+  APPROVED = 'APPROVED',
+  REJECTED = 'REJECTED',
+  PENDING = 'PENDING',
+  COMPLETED = 'COMPLETED',
+}
+
+export const ApprovalStatus = z.enum([
+  WorkflowStatus.APPROVED,
+  WorkflowStatus.REJECTED,
+])
+
+const SubmittedNonApprovalStep = z.object({
+  isApproval: z.literal(false),
+  submittedAt: z.string(),
```

Comment:
does zod have a timestamp type for validation? if so, maybe we can use it

Reply:
yes - fixed referring to [zod's datetime documentation](https://zod.dev/?id=datetimes)